sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -705,10 +713,11 @@ def __init__(self,
     self._key_coder = key_coder
     self._window_coder = window_coder
     # A mapping of {timer_family_id: TimerInfo}
-    self._timers_info = {}
-    self._all_states = {}  # type: Dict[tuple, userstate.RuntimeState]
+    self._timers_info = {}  # type: Dict[str, TimerInfo]
+    self._all_states = {}  # type: Dict[tuple, FnApiUserRuntimeStateTypes]
```

Comment:
`userstate.RuntimeState` does not have the necessary `commit()` method, and there is no common base class for these 4 types that has all of the necessary methods.  When dealing with a reasonably limited number of possibilities with unclear parentage, a `Union` is a good solution.